flotilla-client/main.go

```diff
@@ -49,11 +52,32 @@ func main() {
 		numMessages  = flag.Uint("num-messages", defaultNumMessages, "number of messages to send from each producer")
 		messageSize  = flag.Uint64("message-size", defaultMessageSize, "size of each message in bytes")
 		startupSleep = flag.Uint("startup-sleep", defaultStartupSleep, "seconds to wait after broker start before benchmarking")
+		coordinator  = flag.String("coordinator", "", "http ip & port address for etcd")
+		flota        = flag.String("flota", "", "test group the deamon is part of")
+		numdaemons   = flag.Int("num-daemons", defaultNumDaemon, "The number of daemons the coordinator should wait for before starting")
 	)
 	flag.Parse()
 
 	peers := strings.Split(*peerHosts, ",")
 
+	// If we are to use etcd then start the cluster coordination
+	var cclient coordinate.Client
+	if coordinator != nil && *flota != "" && *numdaemons >= 0 {
+		log.Printf("Starting Cluster for %s and waiting for %d daemons", *flota, *numdaemons)
+		cclient := coordinate.NewSimpleCoordinator(*coordinator, *flota)
+		up, err := cclient.StartCluster(*numdaemons, int(*startupSleep))
+		if up {
+			log.Println("Cluster Started")
+		} else {
+			log.Println("Cluster did not start")
+			if err != nil {
+				log.Println("Cluster start up error was ", err.Error())
+			}
+			os.Exit(1)
+		}
+		peers = append(peers, cclient.ClusterMembers()...)
```

Comment:
Do we need to de-dupe peers? I.e. if the user passes in some specific peers and they overlap with the cluster.